lib/events.ts

```diff
@@ -216,7 +215,7 @@ export const wait = async (
 	jellyfish: JellyfishKernel,
 	session: string,
 	options: WaitOptions,
-): Promise<ExecuteContract> => {
+): Promise<ExecuteContract | undefined> => {
```

Comment:
We have been unknowingly (?) returning `undefined` for a while now because of Bluebird. See `resolve(result || undefined);` below. Want to get a second opinion on this result type change before merging. @LucianBuzzo @Hades32

Reply:
@joshbwlng is that really the case? I'm not familiar with `Stream`, but I had expected that the `close` event would only fire when it's actively closed (and that we'd get an `error` event when the server closes the connection). If that's the case, shouldn't we then always get a result?

And if that's not the case, shouldn't we `reject` if the result hasn't been defined on closing?

Reply:
@Hades32 You may be right in that we were never actually returning `undefined`, I didn't dig that deep. It could be a case in which using Bluebird allowed for improper values in its `resolve()` and we just never removed the `|| undefined` bit.

Reply:
Pushed another commit that removes `undefined` from possible return types in favor of `ExecuteContract` alone.